File: 169-majority-element/majority-element.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        std = sorted(nums)
        maxrun = 0
        currun = 0
        maxel = 0
        prevel = std[0]
        for n in std:
            if n == prevel:
                currun += 1
            else:
                if (currun > maxrun):
                    maxrun = currun
                    maxel = prevel
                currun = 1
            prevel = n
        if (currun > maxrun):
            maxel = prevel
        return maxel
        

File: 169-majority-element/majority-element.c
int majorityElement(int* nums, int numsSize) {
    int cand = 0;
    int count = 0;
    for (int i = 0; i < numsSize; i++) {
        cand = (count == 0) ? nums[i] : cand;
        count = (nums[i] == cand) ? count + 1 : count - 1;
    }
    return cand;
}